### PR TITLE
Fix Datepicker range display

### PR DESCRIPTION
**Pull request by mickaelzhang**
## Screenshot / video

[if applicable for UI changes]

## What does this do?

[A short description of what the PR changes and why it's necessary]

- Give the reviewers context
- Is the impact global or localised?
- Any tech debt created or discovered?
- Include links to Jira ticket and Figma file if applicable

### CODE REVIEW

File: src/Datepicker/Datepicker.tsx [hunk 1]
```diff
@@ -131,31 +91,29 @@ export const Datepicker: FC<DatepickerProps> = ({
         direction="row"
       >
         <Circle
-          disabled={listDays[activeMonth].monthName === listDays[0].monthName}
-          onClick={() => setActiveMonth(activeMonth - 1)}
+          disabled={activeMonthIndex === 0}
+          onClick={() => setActiveMonth(activeMonthIndex - 1)}
```
Comment (mickaelzhang) [inline src/Datepicker/Datepicker.tsx:95]: We were comparing string before but we can compare the index directly instead which would achieve the same

File: src/Datepicker/Datepicker.tsx [hunk 2]
```diff
@@ -84,44 +74,14 @@ export const Datepicker: FC<DatepickerProps> = ({
...
-        endDay,
-      )
-
-      availableDays = [thisMonth, nextMonth]
-    } else {
-      availableDays = [generateDaysForMonth(today, endDay)]
-    }
-
-    setListDays(availableDays)
-  }
```
Comment (mickaelzhang) [inline src/Datepicker/Datepicker.tsx:119]: We were calculating all month before, I switched to only generate the days of the current displayed month

File: src/Datepicker/Datepicker.tsx [hunk 3]
```diff
@@ -22,7 +19,18 @@ import { Icon } from '../Icon'
...
+  const monthList = eachMonthOfInterval({
+    start: startDate,
+    end: endDate,
+  })
+
+  return monthList.map((monthDate) => ({
+    date: monthDate,
+    label: format(monthDate, 'MMMM'),
+  }))
+}
```
Comment (mickaelzhang) [inline src/Datepicker/Datepicker.tsx:33]: Only the months are generated in advance, although there is probably a case to be made for this to be generated progressively as well
For case where there is no range for example, although it doesn't seems it's possible with the component today

File: src/Datepicker/DatesList.tsx [hunk 4]
```diff
@@ -1,29 +1,46 @@
...
+  items,
+  handleDateSelect,
+  firstDayShift,
+}) => {
+  return (
+    <Container>
+      {firstDayShift &&
+        Array(getBlankDaysCount(items[0].date))
+          .fill(null)
+          .map((_, index) => <ListItem key={`blankItem-${index}`} disabled />)}
```
Comment (mickaelzhang) [inline src/Datepicker/DatesList.tsx:30]: I moved the blank day (calendar display mode) inside the list, I don't think it made sense to be inside the day generation function as it's not related to the data itself, only presentational

File: src/Datepicker/Datepicker.tsx [hunk 5]
```diff
@@ -39,42 +47,24 @@ export const Datepicker: FC<DatepickerProps> = ({
   fromDate = new Date(),
```
Comment (fleonard) [inline src/Datepicker/Datepicker.tsx:47]: Were you planning to solve the timezone issue in signup? I'm wondering if might be better to set it by default to generate UK time only start dates

**Review by fleonard — APPROVED** [review]
LGTM, just a minor comment about timezones

File: src/Datepicker/Datepicker.tsx (hunk 5 above)
Comment (mickaelzhang) [reply src/Datepicker/Datepicker.tsx:47]: Yeah, planning to fix the time zone issue on a separate PR! Will open the PR this morning

**Pull request merged**